File: contents/bubble_sort/code/php/bubble_sort.php
<?php
declare(strict_types=1);

function bubble_sort(array $arr): array
{
    for ($i = 0, $length = count($arr); $i < $length; $i++) {
        for ($j = 1; $j < $length; $j++) {
            if ($arr[$j - 1] > $arr[$j]) {
                $tmp = $arr[$j - 1];
                $arr[$j - 1] = $arr[$j];
                $arr[$j] = $tmp;
            }
        }
    }

    return $arr;
}

$unsorted = [1, 2, 6, 47, 4, 9, 3, 7, 8, 23, 15];
$bubble_sorted = bubble_sort($unsorted);

printf('Unsorted: %s', implode(',', $unsorted));
echo PHP_EOL;
printf('Sorted: %s', implode(',', $bubble_sorted));
echo PHP_EOL;

File: contents/bubble_sort/code/dart/bubble_sort.dart
void main() {
  List<int> list = [5, 1, 4, 2, 8];
  List<int> sortedarray = bubbleSort(list);
  print(sortedarray);
}



bubbleSort(List<int> list) {
  var sorted = false;
  var counter = 0;

  while (!sorted) {
    sorted = true;
    for (var i = 0; i < list.length - 1 - counter; i++) {
      if (list[i] > list[i + 1]) {
        final tmp = list[i];
        list[i] = list[i + 1];
        list[i + 1] = tmp;
        sorted = false;
      }
    }
    counter++;
  }
  return list;
}

